internal/k8s/service/resolver.go

```diff
@@ -55,80 +55,48 @@ func NewRefNotPermittedError(inner string) ResolutionError {
 	return ResolutionError{inner, RefNotPermittedErrorType}
 }
 
+func getErrorTypePrefix(errType ServiceResolutionErrorType) string {
+	switch errType {
+	case K8sServiceResolutionErrorType:
+		return "k8s: "
+	case ConsulServiceResolutionErrorType:
+		return "consul: "
+	default:
+		return ""
+	}
+}
+
 func (r ResolutionError) Error() string {
 	return r.inner
 }
 
 type ResolutionErrors struct {
-	k8sErrors             []ResolutionError
-	consulErrors          []ResolutionError
-	genericErrors         []ResolutionError
-	refNotPermittedErrors []ResolutionError
+	errors map[ServiceResolutionErrorType][]ResolutionError
```

Comment:
This might be pulling at a thread that unravels an entire sweater (and would require significantly more refactoring), but I'm curious if there's actually a reason to break up these errors by map keys, as opposed to something like directly using
```
    errrors []ResolutionError
```
instead of this entire struct, and adding fields like
```
type ResolutionError struct {
    origin: ConsulErrorType | K8sErrorType
    type: ServiceResolutionError | RefNotPermittedError
```